From 9bcfc25cb5b46f1eb8ed4bd6312fac4bcdab43a8 Mon Sep 17 00:00:00 2001
From: kashike <kashike@vq.lc>
Date: Thu, 1 Sep 2016 09:51:31 +0000
Subject: [PATCH] Convert new health to a float during set

Convert the new health value to a float before doing any validation logic

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
index b807a3f..9e19e7c 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
@@ -91,6 +91,7 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
     }
 
     public void setHealth(double health) {
+        health = (float) health; // Paper - convert health to a float during set to avoid the below error
         if ((health < 0) || (health > getMaxHealth())) {
             // Paper - Be more informative
             throw new IllegalArgumentException("Health must be between 0 and " + getMaxHealth() + ", but was " + health
-- 
2.9.3

